src/go/rpk/pkg/cli/cluster/connections/list.go

```diff
@@ -0,0 +1,210 @@
+}
+
+func buildFilters(fset *pflag.FlagSet) ([]string, error) {
+	clauses := []string{}
+
+	state := fset.Lookup("state")
+	if state.Changed {
+		val, ok := stateOptionMap[state.Value.String()]
+		if !ok {
+			return nil, fmt.Errorf("%w: %s", errInvalidStateFilter, state.Value.String())
+		}
+		clauses = append(clauses, fmt.Sprintf("state = %s", val.String()))
+	}
+
+	// We have a series of string options we need to map to the upstream API filters
+	for ours, theirs := range map[string]string{
+		"ip-address":                    "source.ip_address",
+		"client-id":                     "client_id",
+		"client-software-name":          "client_software_name",
+		"client-software-version":       "client_software_version",
+		"group-id":                      "group_id",
+		"authentication-user-principal": "authentication_info.user_principal",
+	} {
+		flag := fset.Lookup(ours)
+		if flag.Changed {
+			clauses = append(clauses, fmt.Sprintf("%s = %q", theirs, flag.Value.String()))
+		}
+	}
+
+	idleFlag := fset.Lookup("idle-ms")
+	if idleFlag.Changed {
+		idleMs, err := fset.GetInt64("idle-ms")
+		if err != nil {
+			return nil, err
+		}
+
+		clauses = append(clauses, fmt.Sprintf("idle_duration > %dms", idleMs))
+	}
+	return clauses, nil
+}
+
+func newConnectionList(fs afero.Fs, p *config.Params) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "Display statistics about current connections",
+		Long:  `Display statistics about current connections.`,
+		Args:  cobra.ExactArgs(0),
+		Run: func(cmd *cobra.Command, _ []string) {
+			prof, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "rpk unable to load config: %v", err)
+			cl, err := adminapi.NewClient(cmd.Context(), fs, prof)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			// Build the filters based on the current flag set
+			filterClauses, err := buildFilters(cmd.Flags())
+			out.MaybeDie(err, "invalid filters: %v", err)
+
+			// TODO: add guardrails and define a proper field mapping
+			orderBy := ""
+			orderByFlag := cmd.Flags().Lookup("order-by")
+			if orderByFlag.Changed {
+				orderBy = orderByFlag.Value.String()
+			}
+
+			// TODO: establish a limit
+			limit, err := cmd.Flags().GetUint32("limit")
+			out.MaybeDie(err, "Invalid limit: %s", err)
+
+			req := adminv2.ListKafkaConnectionsRequest{
+				Filter:   strings.Join(filterClauses, " AND "),
+				OrderBy:  orderBy,
+				PageSize: int32(limit),
+			}
+
+			svc := cl.BrokerService()
+			resp, err := svc.ListKafkaConnections(cmd.Context(), &connect.Request[adminv2.ListKafkaConnectionsRequest]{Msg: &req})
+			out.MaybeDie(err, "error listing connections: %v", err)
+
+			if p.Formatter.IsText() {
+				fmt.Println(printConnectionListTable(resp.Msg))
+			} else {
+				_, _, output, err := p.Formatter.Format(resp.Msg.Connections)
+				out.MaybeDie(err, "error listing connections: %v", err)
+				fmt.Println(output)
+			}
+		},
+	}
+
+	// Set filtering flags
+	fset := cmd.Flags()
```

Comment:
Let's use the TypeVarP flags to assign the value of a flag to a variable and avoid using `limit, err := cmd.Flags().GetUint32("limit")` and such in the function execution nor drilling down the flags to other functions.

Examples:
- Normal/a few variables: https://github.com/redpanda-data/redpanda/blob/dev/src/go/rpk/pkg/cli/registry/schema/create.go#L107
- When there are many variables that work with each other we tend to create a type and add methods to it, this seems to be the case: https://github.com/redpanda-data/redpanda/blob/dev/src/go/rpk/pkg/cli/shadow/status.go#L58